test/unit/org/apache/cassandra/repair/RepairTrackerTest.java

```diff
@@ -0,0 +1,128 @@
+    private RepairTracker tracker;
+
+    @Test
+    public void getRepairState_unknownIfCommandNotTracked()
+    {
+        tracker = new RepairTracker();
+        assertThat(tracker.getRepairState(1)).isEqualTo(StorageServiceMBean.ProgressState.UNKNOWN);
+    }
+
+    @Test
+    public void getRepairState_returnsExpected()
+    {
+        tracker = new RepairTracker();
+        RepairRunnable task = mock(RepairRunnable.class);
+        doReturn(StorageServiceMBean.ProgressState.FAILED).when(task).getCurrentState();
+        tracker.track(1, args, task);
+
+        RepairRunnable task2 = mock(RepairRunnable.class);
+        doReturn(StorageServiceMBean.ProgressState.SUCCEEDED).when(task2).getCurrentState();
+        tracker.track(2, args, task2);
+
+        assertThat(tracker.getRepairState(1)).isEqualTo(StorageServiceMBean.ProgressState.FAILED);
+        assertThat(tracker.getRepairState(2)).isEqualTo(StorageServiceMBean.ProgressState.SUCCEEDED);
+    }
+
+    @Test
+    public void getInProgressRepair_emptyWhenNotTracked()
+    {
+        tracker = new RepairTracker();
+        RepairArguments args = mock(RepairArguments.class);
+        assertThat(tracker.getInProgressRepair(args)).isEmpty();
+    }
+
+    @Test
+    public void getInProgressRepair_onlyReturnsInProgress()
+    {
+        tracker = new RepairTracker();
+        RepairRunnable task = mock(RepairRunnable.class);
+        doReturn(1).when(task).getCommand();
+        tracker.track(1, args, task);
+
+
+        Set<StorageServiceMBean.ProgressState> nonInProgress = Arrays.stream(StorageServiceMBean.ProgressState.values())
+                                                 .filter(state -> StorageServiceMBean.ProgressState.IN_PROGRESS != (state))
+                                                 .collect(Collectors.toSet());
+
+        for (StorageServiceMBean.ProgressState state : nonInProgress)
+        {
+            doReturn(state).when(task).getCurrentState();
+            assertThat(tracker.getInProgressRepair(args)).isEmpty();
+        }
+
+        doReturn(StorageServiceMBean.ProgressState.IN_PROGRESS).when(task).getCurrentState();
+        assertThat(tracker.getInProgressRepair(args)).contains(1);
+    }
+
+    @Test
+    public void getInProgressRepair_onlyReturnsMostRecentlyTracked()
+    {
+        tracker = new RepairTracker();
+        RepairRunnable task1 = mock(RepairRunnable.class);
+        doReturn(1).when(task1).getCommand();
+        doReturn(StorageServiceMBean.ProgressState.IN_PROGRESS).when(task1).getCurrentState();
+        tracker.track(1, args, task1);
+
+        RepairRunnable task2 = mock(RepairRunnable.class);
+        doReturn(StorageServiceMBean.ProgressState.IN_PROGRESS).when(task2).getCurrentState();
+        doReturn(2).when(task2).getCommand();
+        tracker.track(2, args, task2);
+
+        assertThat(tracker.getInProgressRepair(args)).contains(2);
+    }
+
+    @Test
+    public void getInProgressRepair_matchesSameArgs()
```

Comment:
Could we also test that it doe not return when the repair args are different? I think it'd also be worth while trying a couple of different permutations of the repair args, to ensure your hashcode/equals works as expected!